enhancements/microshift/optionally-deploy-csi-plugin.md

```diff
@@ -0,0 +1,209 @@
+---
+title: optionally-deploy-csi-plugin
+authors:
+  - "@copejon"
+reviewers:
+  - "@pacevedom: MicroShift team-lead"
+  - "@jerpeter1, Edge Enablement Staff Engineer"
+  - "@jakobmoellerdev Edge Enablement LVMO SME"
+  - "@suleymanakbas91,  Edge Enablement LVMO SME"
+approvers:
+  - "@jerpeter1, Edge Enablement Staff Engineer"
+api-approvers:
+  - "None"
+creation-date: 2024-06-10
+last-updated: 2024-06-11
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-856
+see-also:
+  - "enhancements/microshift/microshift-default-csi-plugin.md"
+---
+
+# Optionally Deploy CSI Plugin
+
+## Summary
+
+MicroShift is a small form-factor, single-node OpenShift targeting IoT and Edge Computing use cases characterized by
+tight resource constraints, unpredictable network connectivity, and single-tenant workloads. See
+[kubernetes-for-devices-edge.md](./kubernetes-for-device-edge.md) for more detail.
+
+Out of the box, MicroShift includes the LVMS CSI driver and CSI snapshotter. The LVM Operator is not itself packaged
+with the platform, in adherence with the project's [guiding principles](./kubernetes-for-device-edge.md#goals). See the
+[MicroShift Default CSI Plugin](microshift-default-csi-plugin.md) proposal for a more in-depth explanation of the
+storage provider and reasons for its integration into MicroShift. Configuration of the driver is exposed via a config
+file at /etc/microshift/lvmd.yaml. The manifests required to deploy and configure the driver and snapshotter are baked
+into the MicroShift binary during compilation and are deployed during runtime. This means that even if a user wanted to
+disable the driver and/or snapshotter, they would be deployed again when the service restarts. Additionally, the LVMS an
+CSI containers required to run LVMS are always packaged with the product and thus will always consume a certain amount
+of storage space devices where MicroShift is deployed.
+
+## Motivation
+
+Not all users may persistent storage provider or volume snapshot capabilities for their MicroShift deployments and
+should be enabled to choose whether to deploy the driver and / or snapshotter. This will afford resource-conscious users
+the opportunity to better tune their MicroShift deployment to their hardware requirements.
+
+### User Stories
+
+- A user deploys MicroShift with the LVMS driver and CSI snapshot controller.
+- A user deploys MicroShift with the LVMS driver only.
+- A user deploys MicroShift with neither the LVMS driver nor the CSI snapshot controller.
+
+### Goals
+
+- Provide the LVMS CSI driver and LVMS CSI snapshot controller as optional components of MicroShift
+- Reuse the optional-installation pattern implemented by MicroShift for the [Multus CNI
+  Plugin](./multus-cni-for-microshift.md).
+- Enable installing the LVMS CSI driver and LVMS CSI snapshot controller to a cluster that was previously deployed
```

Comment:
This last point should already be possible today after a restart right? not sure how this would be differing, could you elaborate a bit more?

Reply:
That's correct. This line is essentially saying it also should be possible in the proposed installation process. See line 107 for steps to install on an existing microshift cluster.

Reply:
gotcha, I guess then im wondering if its a "goal"? Maybe the goal is to preserve that capability in the new approach?

Reply:
That's true, it's not a good fit as a goal, will change